rules/composer/src/ValueObject/AddOrChangePackage.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Composer\ValueObject;
+
+use Rector\Composer\Modifier\ComposerModifier;
+use Rector\Composer\Contract\ComposerModifier\ComposerModifierInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Adds package to selected section, if package is already in composer data, package is updated (version and/or section)
+ * @see \Rector\Composer\Tests\ValueObject\AddOrChangePackageTest
+ */
+final class AddOrChangePackage implements ComposerModifierInterface
+{
+    /** @var string */
+    private $packageName;
+
+    /** @var string */
+    private $version;
+
+    /** @var string */
+    private $section;
+
+    /**
+     * @param string $packageName name of package (vendor/package)
+     * @param string $version target package version (1.2.3, ^1.2, ~1.2.3 etc.)
+     * @param string $section require or require-dev
+     */
+    public function __construct(string $packageName, string $version, string $section = ComposerModifier::SECTION_REQUIRE)
+    {
+        Assert::oneOf($section, [ComposerModifier::SECTION_REQUIRE, ComposerModifier::SECTION_REQUIRE_DEV]);
+
+        $this->packageName = $packageName;
+        $this->version = $version;
```

Comment:
It would be great if version is validated here and standardized as an object. We have a `VersionFactory` to check this, or `Schema` service from composer can be used

Reply:
I am trying to find one of these classes in Rector (or vendor), but no success. I just found `PhpVersionFactory` and `Composer\Semver\VersionParser` with method `normalize()`, but it will be just some validation and it will not create version as an object. Can you navigate me to classes you have in mind?

Reply:
Something like this: https://github.com/symplify/symplify/blob/master/packages/monorepo-builder/src/ValueObjectFactory/VersionFactory.php

It might require combination or new object. The goal is object API, not a string, so it's reliable

Reply:
done